server/appbackend/serializers.py

```diff
@@ -46,6 +48,7 @@ class ClassGroupSerializer(serializers.ModelSerializer):
     teacher = serializers.ReadOnlyField(source='teacher.first_name')
     collaborators = CollaboratorSerializer(many=True, read_only=True)
     course = serializers.ReadOnlyField(source='course.name')
+    semester = serializers.CharField(source='semester.name')
```

Comment:
Semester is an Integer, perhaps deserializing it as charfield is causing the error

Reply:
It isn't.

Reply:
By the way, the problem is in Enrollment, this is ClassGroup